Source/WebKit/GPUProcess/graphics/RemoteImageBufferSet.cpp

```diff
@@ -231,12 +231,17 @@ void RemoteImageBufferSet::prepareBufferForDisplay(const WebCore::Region& dirtyR
     m_frontBufferIsCleared = false;
 }
 
-bool RemoteImageBufferSet::makeBuffersVolatile(OptionSet<BufferInSetType> requestedBuffers, OptionSet<BufferInSetType>& volatileBuffers)
+bool RemoteImageBufferSet::makeBuffersVolatile(OptionSet<BufferInSetType> requestedBuffers, OptionSet<BufferInSetType>& volatileBuffers, bool forcePurge)
 {
     bool allSucceeded = true;
 
-    auto makeVolatile = [](ImageBuffer& imageBuffer) {
+    auto makeVolatile = [&](ImageBuffer& imageBuffer) {
```

Comment:
Could this set of trampolined function calls just be inlined in the caller?
```
bool allSucceeded = false;
for (auto bufferInSet : requestedBuffers) {
    RefPtr<ImageBuffer> buffer;
    if (bufferInSet == BufferInSetType::Front)
       buffer = m_frontBuffer;
    else if (bufferInSet == BufferInSetType::Back)
       buffer = m_backBuffer;
    else if (bufferInSet == BufferInSetType::SecondaryBack)
       buffer = m_secondaryBackBuffer;
    if (!buffer)
       continue;
    buffer->releaseGraphicsContext();
    bool didPurge = false;
    if (purge)
       didPurge = buffer->setVolatileAndPurgeForTesting();
    else
       didPurge = buffer->setVolatile();
    if (didPurge)
       volatileBuffers.add(bufferInSet);
     else
       allSucceeded = false;
}
```

Reply:
It could! Just a personal preference as to whether that's an improvement or not :)